scripts: remove board name from the identifier in list_boards.py

tejlmand opened the pull request:
Fixes: #69329

The board name was printed as part of printing board identifiers because those was being concatenated in the `board_v2_identifiers()` method.

Keep the identifiers speparated from the board name and let the caller concatenate the strings when required.

tejlmand committed:
scripts: remove board name from the qualifiers in list_boards.py

Fixes: #69329

The board name was printed as part of printing board qualifiers because
those was being concatenated in the `board_v2_qualifiers()` method.

Keep the qualifiers separated from the board name and let the caller
concatenate the strings when required.

Completion scripts are also updated to handle the corrected behaviour.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

nordicjm reviewed (changes requested):
west completion files for bash/fish/zsh need updating to account for this

57300 commented:
I guess this should be updated in light of #69905 to print `qualifiers`, or revert to printing what are now called `targets`.

tejlmand commented:
> I guess this should be updated in light of #69905 to print `qualifiers`, or revert to printing what are now called `targets`.

to avoid n-PRs updating terminology, then the plan is a single PR updating all occurrences, hence the reason this PR does not address the terminology issue in existing places (identifiers --> qualifiers).

Will be done in dedicated PR.

nashif commented on scripts/west_commands/boards.py:
```diff
@@ -49,8 +49,9 @@ def do_add_parser(self, parser_adder):
             - name: board name
-            - identifiers: board identifiers
-            - arch: board architecture
+            - identifiers: board identifiers (will be empty for legacy boards)
```
we do not call this identifiers anymore, no?

Another thing, `west boards -f "{identifiers}"`  gives me a list of qualifiers, if a board has multiple, they are joined by a comma, so if I want to get the list of targets using "-f", it becomes very difficult. How about adding `{targets}`

nashif replied:
ok, read the comment above re qualifiers, but rest of the above still applies.

tejlmand replied:
> so if I want to get the list of targets using "-f", it becomes very difficult. How about adding {targets}

yes, but in this initial PR i didn't want to trigger this discussion again related to legacy vs new model boards:
https://github.com/zephyrproject-rtos/zephyr/pull/69740#discussion_r1522772483

Therefore this initial PR which simply ensures that issues are fixed and completion works.
Then a second PR allowing for longer discussion to take place.

I don't like to make an unnecessary block on the fix in this PR.

tejlmand commented:
The compliance error:
``` R0801:Similar lines in 2 files
File:scripts/west_commands/runners/openocd.py
Line:1
Column:0
 R0801:Similar lines in 2 files
```
happens because a lot of duplicated code already exists between https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/intel_cyclonev.py and https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/openocd.py

tejlmand commented:
> The compliance error:

Add `# pylint: disable=duplicate-code`

nordicjm commented:
Rebase needed, documentation build fails and needs fixing

nordicjm commented on scripts/west_commands/completion/west-completion.fish:
```diff
@@ -196,19 +196,12 @@ function __zephyr_west_complete_help
-        for variant in $b
-            printf "%s\n" $variant[1]
+        set -l b (string split "|" $board)
+        set -l identifiers (string split "," $b[2])
```
fish autoocomplete does not work (just blinks and does nothing)

nordicjm commented on scripts/west_commands/completion/west-completion.zsh:
```diff
@@ -102,9 +102,12 @@ _get_west_projs() {
-  _west_boards=${_west_boards//,/\ }
+  _west_boards=( $(__west_x boards --format='{name}|{identifiers}') )
+  for i in {1..${#_west_boards[@]}}; do
+    local name="${_west_boards[$i]%%|*}"
```
zsh autocomplete does not work: `_get_west_boards:5: _west_boards: assignment to invalid subscript range`

nordicjm commented on scripts/west_commands/completion/west-completion.bash:
```diff
@@ -392,8 +392,13 @@ __set_comp_west_projs()
-	boards="$(__west_x boards --format={identifiers} "$@")\n$(__west_x boards --format={name} "$@")"
-	__set_comp ${boards//,/\ }
+	boards=( $(__west_x boards --format='{name}|{identifiers}' "$@") )
+	for i in ${!boards[@]}; do
```
bash autocomplete does not work

nordicjm reviewed (changes requested):
Testing with PR rebased on `main`

nordicjm commented:
Thinking about it, board revisions should probably be present in the autocomplete names too

tejlmand commented:
> Testing with PR rebased on `main`

Rebased and adjusted to use qualifiers instead of identifiers.
The errors you noticed was due to the `--format` string was not adjusted to match the changes in #70438

tejlmand commented:
> Thinking about it, board revisions should probably be present in the autocomplete names too

That's a good idea, and I think there was a similar comment not long ago, but forgot where it was.

Although the idea is good, and could be implemented for know revisions (those provided in board.yml), then we must consider how we want to handle `custom` revisions, as those may not be present in the board.yml.

This is a new possibility with HWMv2, which we couldn't do in HWMv1 because revisions was handled purely within CMake.

tejlmand commented:
> This is a new possibility with HWMv2, which we couldn't do in HWMv1 because revisions was handled purely within CMake.

Created https://github.com/zephyrproject-rtos/zephyr/issues/70597

57300 reviewed (approved):
LGTM, tested bash autocompletion.
